NuclearData/xsPackages/IMCXsPackages_class.f90

```diff
@@ -0,0 +1,255 @@
+  !!   inelasticScatter -> sum of all IMC producing reaction that are not elastic scattering [1/cm]
+  !!   capture          -> sum of all reactions without secondary photons [1/cm]
+  !!
+  !!  Interface:
+  !!    clean -> Set all XSs to 0.0
+  !!    add   -> Add a nuclide microscopic XSs to macroscopic
+  !!    get   -> Return XS by MT number
+  !!
+  type, public :: IMCMacroXSs
+    real(defReal) :: total            = ZERO
+    real(defReal) :: elasticScatter   = ZERO
+    real(defReal) :: inelasticScatter = ZERO
+    real(defReal) :: capture          = ZERO
+  contains
+    procedure :: clean => clean_IMCMacroXSs
+    procedure :: add   => add_IMCMacroXSs
+    procedure :: get
+    procedure :: invert => invert_macroXSs
+  end type IMCMacroXSs
+
+
+  !!
+  !! IMC microscopic Reaction XSS
+  !!
+  !! Public Members:
+  !!   total            -> total Cross-Section [barn]
+  !!   elasticScatter   -> MT=2 elastic IMC scattering [barn]
+  !!   inelasticScatter -> all photon producing reaction that are not elastic scattering [barn]
+  !!   capture          -> all reactions without secendary photons [barn]
+  !!
+  type, public :: IMCMicroXSs
+    real(defReal) :: total            = ZERO
+    real(defReal) :: elasticScatter   = ZERO
+    real(defReal) :: inelasticScatter = ZERO
+    real(defReal) :: capture          = ZERO
+  contains
+    procedure :: invert => invert_microXSs
+  end type IMCMicroXSs
+
+contains
+
+  !!
+  !! Clean IMC MacroXSs
+  !!
+  !! Sets all XSs to 0.0
+  !!
+  !! Args:
+  !!   None
+  !!
+  !! Errors:
+  !!   None
+  !!
+  elemental subroutine clean_IMCMacroXSs(self)
+    class(IMCMacroXSs), intent(inout) :: self
+
+    self % total            = ZERO
+    self % elasticScatter   = ZERO
+    self % inelasticScatter = ZERO
+    self % capture          = ZERO
+
+  end subroutine clean_IMCMacroXSs
+
+  !!
+  !! Add nuclide XSs on Macroscopic XSs
+  !!
+  !! Takes microscopic XSs * density and adds them to IMCMacroXSs
+  !!
+  !! Args:
+  !!   micro [in] -> microscopic XSs
+  !!   dens  [in] -> nuclide density in [1/barn/cm]
+  !!
+  !! Errors:
+  !!   None
+  !!
+  elemental subroutine add_IMCMacroXSs(self, micro, dens)
+    class(IMCMacroXSs), intent(inout) :: self
+    type(IMCMicroXSs), intent(in)     :: micro
+    real(defReal), intent(in)             :: dens
+
+    self % total            = self % total            + dens * micro % total
+    self % elasticScatter   = self % elasticScatter   + dens * micro % elasticScatter
+    self % inelasticScatter = self % inelasticScatter + dens * micro % inelasticScatter
+    self % capture          = self % capture          + dens * micro % capture
+
+  end subroutine add_IMCMacroXSs
+
+  !!
+  !! Return XSs by MT number
+  !!
+  !! Args:
+  !!   MT [in] -> Requested MT number
+  !!
+  !! Result:
+  !!   Value of the XS
+  !!
+  !! Errors:
+  !!   Returns 0.0 for invalid MT
+  !!
+  elemental function get(self, MT) result(xs)
```

Comment:
You can make up a number for the Planck opacity in ENDF constants if you want to get it through here.